src/lib/work_partitioner/job_pool.mli

```diff
@@ -0,0 +1,33 @@
+open Core_kernel
+
+type ('accum, 'final) fold_action =
+  | Continue of 'accum
+  | Continue_remove of 'accum
+  | Stop of 'final
+  | Stop_remove of 'final
+
+module Make (Id : Map.Key) (Spec : T) : sig
+  type job = (Spec.t, Id.t) Snark_work_lib.With_job_meta.t
+
+  module IdMap : module type of Map.Make (Id)
+
+  type t = job IdMap.t
+
+  val peek : job IdMap.t -> job option
+
+  val fold_until :
+       init:'a
+    -> f:('a -> 'b -> ('a, 'c) fold_action)
+    -> finish:('a -> 'c)
+    -> 'b IdMap.t
+    -> 'c * 'b IdMap.t
+
+  val attempt_add :
```

Comment:
```suggestion
  val add :
```

To make it similar to convention used for maps we use. There are normally `add` and `add_exn`.